scripts/cargo-rmc

```diff
@@ -89,6 +64,152 @@ def main():
+                print(f"WARNING: Cannot find configuration toml at expected location: {cargo_toml}")
+                return None
+
+    # Combine args set by config toml and provided on command line
+    def get_combined_args(parser, config_toml):
+        # Extract flags from config_toml
+        toml_args = extract_args_from_toml(config_toml)
+
+        # Parse args from both config_toml and command line
+        # but let command line args override config_toml args
+        combined_args = argparse.Namespace()
+        parser.parse_args(toml_args, namespace=combined_args)
+        parser.parse_args(sys.argv[1:], namespace=combined_args)
+
+        return combined_args
+
+    # Check for conflicting flags
+    def validate(args):
+        rmc.ensure(not (args.crate and args.crate_flag), "Please provide a single crate to verify.")   
+        rmc.ensure(not (args.no_config_toml and args.config_toml), "Incompatible flags: --config-toml, --no-config-toml")
+
+    # Fix up args before returning
+    def post_process(args):
+        # Combine positional and flag argument for input
+        args.crate = get_crate_from_args(args)
+
+        # --quiet overrides --verbose
+        if args.quiet:
+            args.verbose = False
+
+        # Add some CBMC flags by default unless `--no-default-checks` is being used
+        if args.default_checks:
+            rmc.add_selected_default_cbmc_flags(args)
+
+    # Remove "rmc" from arg if invoked `cargo rmc ...`
+    if len(sys.argv) >= 2 and sys.argv[1] == "rmc":
+        del sys.argv[1]
+
+    # Parse command line args to find crate and check for config flags
+    parser = create_parser()
+    cl_args = parser.parse_args()
+    validate(cl_args)
+
+    # Try to find the config_toml based on command line arguments
+    config_toml = get_config_toml(cl_args.no_config_toml, cl_args.config_toml, cl_args)
+
+    if config_toml is not None:
+        # If config_toml is found, combine configuration with command line arguments
+        combined_args = get_combined_args(parser, config_toml)
+        validate(combined_args)
+        post_process(combined_args)
+        return combined_args
+    else:
+        # If config_toml is missing, just use parsed command line arguments
+        post_process(cl_args)
+        return cl_args
+
+# Extract a list of args based on given config toml
+def extract_args_from_toml(path):
+    # Load the flag data from toml
+    data = toml.load(path)
+
+    # Extract the rmc flags from the toml, if any present
+    if rmc not in data or flags not in rmc_data["rmc"]:
+        # If no flags are present, just return none
+        return []
+    rmc_data = data["rmc"]
+    flag_data = rmc_data["flags"]
+
+    # Extract nested flags
+    flags = dict()
+    def find_flags(map):
+        for key in map:
+            if type(map[key]) == dict:
+                find_flags(map[key])
+            else:
+                flags[key] = map[key]
+    find_flags(flag_data)
+
+    # Add flags to args
```

Comment:
so flags are what `cargo rmc` takes, and args are what we're passing to`rmc`?

Reply:
I've generally had in my head that `flags` was the command line input, while `args` is the result of parsing this, though sometimes the distinction gets mixed up. I've fixed up a few places where this was backwards.